api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -553,6 +553,14 @@ message HttpConnectionManager {
   // of `HTTP spec <https://tools.ietf.org/html/rfc3986>`_ and is provided for convenience.
   bool strip_matching_host_port = 39;
 
+  // Determines if the port part should be removed from host/authority header before any processing
+  // of request by HTTP filters or routing. The port would be removed only if request method is not CONNECT.
+  // This affects the upstream host header as well.
+  // Without setting this option, incoming requests with host `example:443` will not match against
+  // route with :ref:`domains<envoy_api_field_config.route.v3.VirtualHost.domains>` match set to `example`. Defaults to `false`. Note that port removal is not part
+  // of `HTTP spec <https://tools.ietf.org/html/rfc3986>`_ and is provided for convenience.
+  bool strip_any_host_port = 42;
```

Comment:
Does it make sense to make it oneof with the above field?  Was wondering about it but did not because it needs to go through oneof_promotion dance - Not sure if it is worth it here.

Reply:
Yeah I think it should be a oneof, but can you use the next major version oneof upgrade? I agree it's not worth doing the deprecation dance for this but marking it as such seems worth while.